e2e/tests/repeater-paginate.test.js

```diff
@@ -0,0 +1,212 @@
+    },
+  ],
+};
+
+const configPagination = `
+  <View>
+    <Repeater on="$images" indexFlag="{{idx}}" mode="pagination" >
+      <Image name="page_{{idx}}" value="$images[{{idx}}].url"/>
+      <RectangleLabels name="labels_{{idx}}" toName="page_{{idx}}">
+        <Label value="Document Title" />
+        <Label value="Document Date" />
+        <Label value="Document Author" background="yellow"/>
+      </RectangleLabels>
+    </Repeater>
+  </View>
+`;
+
+const configScroll = `
+  <View>
+    <Repeater on="$images" indexFlag="{{idx}}" >
+      <Image name="page_{{idx}}" value="$images[{{idx}}].url"/>
+      <RectangleLabels name="labels_{{idx}}" toName="page_{{idx}}">
+        <Label value="Document Title" />
+        <Label value="Document Date" />
+        <Label value="Document Author" background="yellow"/>
+      </RectangleLabels>
+    </Repeater>
+  </View>
+`;
+
+
+const annotations = [
+  {
+    id: 38,
+    result: [
+      {
+        id: "Eg3_P8-ZRu",
+        type: "rectanglelabels",
+        value: {
+          x: 8.247422680412368,
+          y: 54.22647527910686,
+          width: 75.46391752577318,
+          height: 9.090909090909092,
+          rotation: 0,
+          rectanglelabels: ["Document Date"],
+        },
+        origin: "manual",
+        to_name: "page_0",
+        from_name: "labels_0",
+        image_rotation: 0,
+        original_width: 800,
+        original_height: 1035,
+      },
+      {
+        id: "Eg3_P8-ZRu",
+        type: "taxonomy",
+        value: {
+          x: 8.247422680412368,
+          y: 54.22647527910686,
+          width: 75.46391752577318,
+          height: 9.090909090909092,
+          rotation: 0,
+          taxonomy: [["Archaea"]],
+        },
+        origin: "manual",
+        to_name: "page_0",
+        from_name: "categories_0",
+        image_rotation: 0,
+        original_width: 800,
+        original_height: 1035,
+      },
+      {
+        id: "yd33DCnE52",
+        type: "rectanglelabels",
+        value: {
+          x: 22.379603399433428,
+          y: 70.31044654069684,
+          width: 20.396600566572232,
+          height: 22.560672877544462,
+          rotation: 0,
+          rectanglelabels: ["Document Title"],
+        },
+        origin: "manual",
+        to_name: "page_2",
+        from_name: "labels_2",
+        image_rotation: 0,
+        original_width: 600,
+        original_height: 776,
+      },
+      {
+        id: "7qxBZWOXXG",
+        type: "rectanglelabels",
+        value: {
+          x: 33.711048158640224,
+          y: 60.01577056744838,
+          width: 23.229461756373937,
+          height: 7.447212406179728,
+          rotation: 0,
+          rectanglelabels: ["Document Author"],
+        },
+        origin: "manual",
+        to_name: "page_1",
+        from_name: "labels_1",
+        image_rotation: 0,
+        original_width: 600,
+        original_height: 776,
+      },
+    ],
+  },
+];
```

Comment:
This one is still out of order

Reply:
Moved that one as well. Curious how it works with the data being out of order of declaration and usage. Typescript and the browser aren't able to continue past seeing that type of thing. Unless webpack's compilation and hoisting is fixing it at some point.